platform/linux-generic/odp_crypto_openssl.c

```diff
@@ -2711,6 +2724,113 @@ int crypto_int(odp_packet_t pkt_in,
 	return 0;
 }
 
+/*
+ * Copy cipher range and auth range from src to dst,
+ * with shifting by dst_offset_shift.
+ */
+static void copy_ranges(odp_packet_t dst,
+			odp_packet_t src,
+			const odp_crypto_generic_session_t *session,
+			const odp_crypto_packet_op_param_t *param)
+{
+	odp_packet_data_range_t c_range = param->cipher_range;
+	odp_packet_data_range_t a_range = param->auth_range;
+	uint32_t shift = param->dst_offset_shift;
```

Comment:
Changed the type to signed in v15 but did not yet add any checks (if we start validating packet op input, it might be good to check the ranges etc and with all operation types. I am thinking about making a separate PR about that).